### PR TITLE
Mark hoisted react constant elements as #__PURE__

### PR DESCRIPTION
**Pull request by Andarist**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | y
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-plugin-transform-react-constant-elements/src/index.js [hunk 1]
```diff
@@ -101,7 +102,11 @@ export default function transformReactConstantElement(api, options) {
         path.traverse(immutabilityVisitor, state);
 
         if (state.isImmutable) {
-          path.hoist();
+          const hoisted = path.hoist();
+          // instanceof NodePath
+          if (hoisted instanceof path.constructor) {
```
Comment (jridgewell) [inline packages/babel-plugin-transform-react-constant-elements/src/index.js]: Why not `NodePath`? Is the `instanceof` even necessary?

Comment (Andarist) [reply packages/babel-plugin-transform-react-constant-elements/src/index.js]: just lazy before hitting the bed and I didn't search from where I could import this

`path.hoist()` can return undefined, that is a guard against that - I guess I could reverse the condition and remove 'dep' on the `instanceof` altogether

Comment (jridgewell) [reply packages/babel-plugin-transform-react-constant-elements/src/index.js]: If it's only `undefined` or a path, why not just truthy check `hoisted`?

```js
if (hoisted) {
  //...
}

Comment (Andarist) [reply packages/babel-plugin-transform-react-constant-elements/src/index.js]: Fixed

**Pull request merged**